app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -0,0 +1,186 @@
+    private EditText habitReasonEditText;
+    private EditText habitDateEditText;
+    private Button confirmNewHabit;
+    private Button cancelNewHabit;
+
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.add_habit_dialog_box, container, false);
+        habitTitleEditText = view.findViewById(R.id.habit_title_edittext);
+        habitReasonEditText = view.findViewById(R.id.habit_reason_edittext);
+        habitDateEditText = view.findViewById(R.id.habit_date_edittext);
+        confirmNewHabit = view.findViewById(R.id.confirmHabit);
+        cancelNewHabit = view.findViewById(R.id.cancelHabit);
+
+        db = FirebaseFirestore.getInstance();
+        /**
+         * Name of database collection.
+         */
+        final CollectionReference collectionReference = db.collection("Habits");
+
+        /**
+         * Closes Dialog box if user presses the cancel button.
+         */
+        cancelNewHabit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Cancel");
+                getDialog().dismiss();
+            }
+        });
+
+        confirmNewHabit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Confirm");
+                /**
+                 * Gets user input and saves to string variables.
+                 */
+                String habitTitleInput = habitTitleEditText.getText().toString();
+                String habitReasonInput = habitReasonEditText.getText().toString();
+                String habitDateInput = habitDateEditText.getText().toString();
+                SimpleDateFormat inputDateFormatter = new SimpleDateFormat("dd-MM-yyyy");
+                Date habitDate = null;
+                try {
+                    habitDate = inputDateFormatter.parse(habitDateInput);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+                /**
+                 * Makes sure user does not enter blank fields,
+                 * and is capped on the number of characters for title and reason.
+                 */
+                HashMap<String, Habit> firebaseData = new HashMap<>();
+                Boolean properTitle = false;
+                Boolean properReason = false;
+                Boolean properDate = false;
+                if ((habitTitleInput.length()>0)&&(habitTitleInput.length()<=20)){
+                    properTitle = true;
+                }
+                if((habitReasonInput.length()>0)&&(habitReasonInput.length()<=30)){
+                    properReason = true;
+                }
+                if(!(habitDateInput.equals(""))){
+                    properDate = true;
+                }
```

Comment:
We can probably do this a better way. Instead of setting boolean flags for each one, just move these conditional statements to enable the action we want here instead, basically:

Instead of:
```
if (someCondition) {
flag = true
}

if (flag == true) {
do stuff
}
```
We can do:
```
if (someCondition) {
just do the stuff
}
```

So the **boolean flags** aren't necessary